File: pet_hospital_backend/src/main/java/com/pethospital/entity/Article.java
package com.pethospital.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("articles")
public class Article {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private String title;
    
    private String category;
    
    private String coverImage;
    
    private String summary;
    
    private String content;
    
    private String author;
    
    private Integer viewCount;
    
    private Integer isPublished; // 1=已发布, 0=草稿
    
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime publishedAt;
    
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    
    
    private LocalDateTime deletedAt;
}

File: pet-hospital-miniprogram/src/api/article.ts
import request from './request'

export interface Article {
    id?: number
    title: string
    category?: string
    coverImage?: string
    summary?: string
    content: string
    author?: string
    viewCount?: number
    isPublished?: number
    publishedAt?: string
    createdAt?: string
    updatedAt?: string
}

export const articleApi = {
    // 获取文章列表
    getList: (params?: { category?: string; page?: number; size?: number }) => {
        return request.get('/api/articles/list', params)
    },

    // 获取文章详情
    getDetail: (id: number) => {
        return request.get(`/api/articles/${id}`)
    }
}
